src/gpu/nvidia/cudnn_pooling.cpp

```diff
@@ -81,33 +80,34 @@ status_t cudnn_pooling_fwd_t::execute(const exec_ctx_t &ctx) const {
     }
 
     return cuda_stream->interop_task([&](::sycl::handler &cgh) {
-        auto src_acc = CTX_IN_ACCESSOR(DNNL_ARG_SRC);
-        auto dst_acc = CTX_OUT_ACCESSOR(DNNL_ARG_DST);
-
-        std::shared_ptr<
-                ::sycl::accessor<uint8_t, 1, ::sycl::access::mode::write>>
-                wkspace_acc;
-        if (!wkspace_st->is_null()) {
-            wkspace_acc = std::make_shared<
-                    ::sycl::accessor<uint8_t, 1, ::sycl::access::mode::write>>(
-                    utils::downcast<sycl::sycl_buffer_memory_storage_t *>(
-                            wkspace_st)
-                            ->buffer()
-                            .template get_access<::sycl::access::mode::write>(
-                                    cgh));
-        }
+        auto arg_src = CTX_IN_SYCL_MEMORY(DNNL_ARG_SRC);
+        auto arg_dst = CTX_OUT_SYCL_MEMORY(DNNL_ARG_DST);
+
+        sycl::sycl_memory_storage_base_t *wkspace_st
```

Comment:
With a suggestion described [here ](https://github.com/oneapi-src/oneDNN/pull/1258/files#r807595388)we could simplify the code as follows:
```cpp
// Can be empty, which is valid.
auto arg_wkspace = CTX_OUT_SYCL_MEMORY(DNNL_ARG_WORKSPACE);
// ...
if (!arg_wkspace.empty()) {
    ws_x = static_cast<uint8_t *>(arg_wkspace.get_native_pointer(ih));
    // ...
}
```

Reply:
I have made the change. Thanks

Reply:
LGTM, thanks!